iidm/iidm-criteria/src/main/java/com/powsybl/iidm/criteria/VoltageInterval.java

```diff
@@ -0,0 +1,224 @@
+         */
+        public Builder setHighBound(double value, boolean closed) {
+            checkValue(value);
+            checkBounds(nominalVoltageLowBound, value, lowClosed, closed);
+            this.nominalVoltageHighBound = value;
+            this.highClosed = closed;
+            return this;
+        }
+
+        protected static void checkValue(double value) {
+            if (Double.isNaN(value) || Double.isInfinite(value) || value < 0) {
+                throw new IllegalArgumentException("Invalid interval bound value (must be >= 0 and not infinite).");
+            }
+        }
+
+        protected static void checkBounds(Double low, Double high, boolean closedLow, boolean closedHigh) {
+            if (low != null && high != null && low > high) {
+                throw new IllegalArgumentException("Invalid interval bounds values (nominalVoltageLowBound must be <= nominalVoltageHighBound).");
+            }
+            double l = low != null ? low : 0;
+            double h = high != null ? high : Double.MAX_VALUE;
+            if (l == h && (!closedLow || !closedHigh)) {
+                throw new IllegalArgumentException("Invalid interval: it should not be empty");
+            }
+        }
+
+        public VoltageInterval build() {
+            if (nominalVoltageLowBound == null && nominalVoltageHighBound == null) {
+                throw new IllegalArgumentException("Invalid interval: at least one bound must be defined.");
+            }
+            return new VoltageInterval(nominalVoltageLowBound, nominalVoltageHighBound, lowClosed, highClosed);
+        }
+    }
+
+    /**
+     * Return a builder to create an {@link VoltageInterval}.
+     *
+     * @return a builder
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * <p>Convenient method to easily create a {@link VoltageInterval} with only a lower bound.</p>
+     * @param value the lower bound of the interval to create (it corresponds to the <code>nominalVoltageLowBound</code> attribute of the interval)
+     * @param valueIncluded is the bound included in the interval (it corresponds to the <code>lowClosed</code> attribute of the interval)
+     * @return an interval
+     */
+    public static VoltageInterval greaterThan(double value, boolean valueIncluded) {
+        return VoltageInterval.builder()
+                .setLowBound(value, valueIncluded)
+                .build();
+    }
+
+    /**
+     * <p>Convenient method to easily create a {@link VoltageInterval} with only a upper bound.</p>
+     * @param value the upper bound of the interval to create (it corresponds to the <code>nominalVoltageHighBound</code> attribute of the interval)
+     * @param valueIncluded is the bound included in the interval (it corresponds to the <code>highClosed</code> attribute of the interval)
+     * @return an interval
+     */
+    public static VoltageInterval lowerThan(double value, boolean valueIncluded) {
```

Comment:
Just a detail here but do you change for `valueIncluded`? Closed was nice.

Reply:
I used this term because I think that when someone says "something lower than a value", it is more natural to indicate if the value is "valid" or included ("or equals") than to say that the corresponding interval is closed.

But the intent behind this term is maybe not that clear.

I will change it to "closed" for consistency.